### PR TITLE
Feature/post as org

### LINKED ISSUE #874: Make organization author if creating post as organization
We need both backend and frontend changes but backend is predominat issue.

We can split out F/E to different ticket if needed in `posts.js` `app.post(`.

Backend we need to accept organization object with `createPostSchema`. If it's set then we need to use that organization as author instead of authenticated user in . 

Frontend we need actually send organization - set here https://github.com/FightPandemics/FightPandemics/blob/7d220104fa870e5f57ee68c0c024f754d4550944/client/src/components/CreatePost/CreatePost.js#L67 - but not actually sent to B/E.

### PR DESCRIPTION
**Pull request by joshmorel**
### What does this pull request aim to achieve? 💯

Resolves #874.

Instead of write away merging @cvaldez98 right away I do want a review here.

But you can merge this into your other branch for organization activity if you want to because this does enable posting as an org. 

@Naraujo13 for B/E - I considered two other approaches:

1. Passing whole organization to B/E and using that. But this could actually enable someone else to post as an org they don't own via curl/postman/etc so we do need to look up the ownerId in the DB to make sure it matches posting userId
2. If `organizationId` is set use "organizations" virtual then look through the array to achieve the similar. This seemed awkward however.

Open to other suggestions

@nr431207 @BMariscal @robinv85 for F/E this seems okay. I think I'm using context here correctly. The name form for the context was a little weird (we also have formData state) and I considered changing it to just organizationsId but maybe more could be included in there later.

We need to change post edit/delete so owners can edit organization posts. I will make a new issue.

- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**README**](https://github.com/FightPandemics/FightPandemics/blob/master/README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`** (need to have cloned the repo not forked).
- [x] This branch is rebased/merged with the **latest master**.
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".

### CODE REVIEW

File: client/src/components/CreatePost/CreatePost.js [hunk 1]
```diff
@@ -82,7 +83,7 @@ const Step2 = ({ user }) => {
 
 const Step3 = ({ onCancel }) => {
   const { currentStep, setCurrentStep, setPostId } = useContext(
-    CreatePostContext,
+    CreatePostContext
```
Comment (idjevm) [inline client/src/components/CreatePost/CreatePost.js:86]: no trailing comma?

Comment (joshmorel) [reply client/src/components/CreatePost/CreatePost.js:86]: Yeah this is an issue when I apply prettier --write via docker & didn't catch this mistake. We have `trailingComma: 'all' in root level .prettier.yml. We've got to find a better way to make this work smoothly in docker - see https://github.com/FightPandemics/FightPandemics/issues/786

Comment (idjevm) [reply client/src/components/CreatePost/CreatePost.js:86]: Ohh interesting though! That explains a lot! @CeesWang is probably not running docker, which is why his code gets reformatted on commit and messes some of his stuff...

File: backend/lib/endpoints/posts.js [hunk 2]
```diff
@@ -222,13 +223,35 @@ async function routes(app) {
...
+      const { organizationId } = postProps;
+      if (organizationId) {
+        const [orgErr, org] = await app.to(User.findById(organizationId));
+        if (orgErr) {
+          req.log.error(userErr, "Failed retrieving organization");
+          throw app.httpErrors.internalServerError();
+        } else if (org === null) {
+          req.log.error(userErr, "Organization does not exist");
+          throw app.httpErrors.forbidden();
+        } else if (org.ownerId.toString() !== userId.toString()) {
```
Comment (Naraujo13) [inline backend/lib/endpoints/posts.js:237]: Instead of converting both to string we could use the `.equals()` from the ObjectId

Comment (idjevm) [reply backend/lib/endpoints/posts.js:237]: so from `org.ownerId.toString() !== userId.toString()` to -> `org.ownerId.equals(userId)`? I also think that would be better. I guess we can revisit later.

Comment (Naraujo13) [reply backend/lib/endpoints/posts.js:237]: Yep, that's correct

**Pull request merged**